features/151-cms-import-export/api.md

```diff
@@ -0,0 +1,562 @@
+- Apply filter when export data.
+
+## Requirements
+
+### Config
+
+- Add record list action
+  - add key `actions` to record list config
+  - for each action item
+    - the client should render a button
+    - there must be a `type` value
+    - value of other keys in the item depend on the type
+- Add record list action, type: import, export
+
+#### Spec
+
+```
+records:
+  {record-type}:
+    list:
+      actions:
+        # Import
+        - type: import
+          name: {import-name}
+          label: {import-display-name}
+          reference_handling: {use-first|throw-error} (Naming TBD, details see below)
+          identifer: {field.name|_id}
+          fields:
+            # normal fields
+            - name:
+              label: (default to name)
+
+            # reference
+            - name:
+              label: (default to name)
+              reference_target:
+              reference_field_name:
+
+        # Export
+        - type: export
+          name: {export-name}
+          label: {export-display-name}
+          fields:
+            # normal fields
+            - name:
+              label: (default to name)
+              format: (Optional)
+
+            # reference fields
+            - name:
+              label: (default to name)
+              format: (Optional)
+              reference_target:
+              reference_field_name:
+              reference_back_reference: (Required for one-to-many with id in referenced record type)
+              reference_via_association_records: (Required for many-to-many)
+
+        - type: export
+          name: {export-name}
+          label: {export-display-name}
+          for_import: {import-name}
+```
+
+Keys:
+
+- import-name
+  - import page route name
+  - reference for export, to generate a export config for this import
+- import-display-name
+  - import button text
+  - import page(s) title
+- export-name
+  - export page route name
+- export-display-name
+  - export button text
+  - export page(s) title
+
+#### Field serialisation and deserialisation
+
+Operation:
+- Import: String value -(serialisation)-> typed data -(save)-> Skygear server
+- Export: String value <-(deserialisation)- typed data <-(query)- Skygear server
+
+In import / export field config, there is no need to specify the type. Instead, server needs to know how to transform between skygear data and string.
+
+##### Type-Format mapping
+
+Server perform (de)serialisation based on a format, and for each Skygear data type, there is a list of supported format.
+
+<table>
+  <thead>
+    <tr>
+      <th>Skygear Types</th>
+      <th>Format</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>String</td>
+      <td>String</td>
+    </tr>
+    <tr>
+      <td>Number</td>
+      <td>Number</td>
+    </tr>
+    <tr>
+      <td>Boolean</td>
+      <td>
+        <ul>
+          <li>True/False (Default)</li>
+          <li>Yes/No</li>
+          <li>1/0</li>
```

Comment:
For all of these options, will we give them a name? Or what value would we expect user will fill-in in yaml?